File: Assignments/Assignment_71/program3.c
// Write a recursive program which Accept String from user and count Number of small characters.
// Input  : HElloWOrLD
// Output : 5

#include<stdio.h>

int Small(char *str)
{
    static int iCount = 0;

    if(*str == '\0')
    {
        return iCount;
    }

    if(*str >= 'a' && *str <= 'z')
    {
        iCount++;
    }

    str++;
        
    Small(str);

    return iCount;
}

int main()
{
    int iRet = 0;
    char arr[20];

    printf("Enter String : \n");
    scanf("%[^\n]",arr);

    iRet = Small(arr);

    printf("%d\n",iRet);

    return 0;
}

File: Assignments/Assignment_72/program3.java
// Write a program which accept matrix from user and return largest number from both the diagonals.

import java.util.*;

class Matrix
{
    public int iRow;
    public int iCol;
    public int Arr[][];

    public Matrix(int a, int b)
    {
        this.iRow = a;
        this.iCol = b;

        Arr = new int[iRow][iCol];
    }

    public void Accept()
    {
        int i = 0, j = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the elements of Matrix");

        for(i = 0; i < iRow; i++)
        {
            System.out.println("Enter the elements of Row : "+ (i+1));

            for(j = 0; j < iCol; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }

        sobj.close();
    }

    public void Display()
    {
        int i = 0, j = 0;

        System.out.println("Elements of Matrix are : ");

        for(i = 0; i < iRow; i++)
        {
            for(j = 0; j < iCol; j++)
            {
                System.out.print(Arr[i][j] + "\t");
            }
            System.out.println();
        }
    }

    public int MaxDiagonal()
    {
        int i = 0, j = 0;
        int iMax = 0; 

        for(i = 0; i < iRow; i++)
        {
            for(j = 0; j < iCol; j++)
            {
                if((i == j) || ((i+j) == (4+1)))
                {
                    if(Arr[i][j] > iMax)
                    {
                        iMax = Arr[i][j];
                    }
                }
            }
        }

        return iMax;
    }
}

class program3
{
    public static void main(String A[])
    {
        int iValue = 0;

        Matrix mobj = new Matrix(4,4);

        mobj.Accept();
        mobj.Display();

        System.out.println("Count of number are : "+ mobj.MaxDiagonal());
    }
}